### DISCUSSION TITLE
Can we add an option to `gatsby-plugin-sharp` NOT to exit during build process even when error happens?

### DISCUSSION THREAD

**Discussion by Jeff-Tian** [gatsbyjs/gatsby#37151] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

My building contains a log of images, somehow some images are not supported by `sharp` (will throw errors). But can we add an option to `gastby-plugin-sharp` NOT to exit when this happens?

When images are not supported by `sharp`, the following error would be met:
<img width="673" alt="image" src="https://user-images.githubusercontent.com/3367820/205287198-2e0c796d-8c60-405a-b680-7e0cb1ca229e.png">

When the above error occurs it hard-breaks the building process.

![image](https://user-images.githubusercontent.com/3367820/205286957-0d37d1c7-a5cf-4a3d-a310-f0f362367f3b.png)

Better to have the option to continue ignoring the failing image processing.

### Reproduction Link

https://github.com/Jeff-Tian/space/actions

### Steps to Reproduce

1. Got to the https://github.com/Jeff-Tian/space/actions
2. Check the failed actions
3. Found the building process exit without any clue

![image](https://user-images.githubusercontent.com/3367820/205287563-30dc9eda-58c5-43d3-9ace-745418466cd0.png)

### Expected Result

The building process continues leaving the failed image processing.

### Actual Result

The building hard stops.

### Environment

```shell
Both on github actions and gatsby cloud
```

### Config Flags

By writing some test cases I found the root cause is the devil `process.exit(1)` code in the `gatsby-plugin-sharp`.

https://stackoverflow.com/questions/74546620/gatsby-remark-images-remote-make-the-build-fail-how-to-troubleshoot-the-root-ca

**Comment by LekoArts** [comment]
Have you used the `failOn` setting in https://www.gatsbyjs.com/plugins/gatsby-plugin-sharp/#options ?

**Reply by Jeff-Tian** [reply]
The `failOn` isn't read by the `gatsby-plugin-sharp` during `reportError`, so I fixed it by #37165 . Adn tested it is now working as expected.

**Comment by Jeff-Tian** [comment]
Yes I set it to none but no effect. I think it should read the option failon before exit

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Lennart ***@***.***>
发送时间: Friday, December 2, 2022 9:55:59 PM
收件人: gatsbyjs/gatsby ***@***.***>
抄送: Jeff Tian ***@***.***>; Author ***@***.***>
主题: Re: [gatsbyjs/gatsby] Can we add an option to `gatsby-plugin-sharp` NOT to exit during build process even when error happens? (Discussion #37151)


Have you used the failOn setting in https://www.gatsbyjs.com/plugins/gatsby-plugin-sharp/#options<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.gatsbyjs.com%2Fplugins%2Fgatsby-plugin-sharp%2F%23options&data=05%7C01%7C%7C24937de547524a1b355808dad46cf18b%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638055861642904943%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=IniaVaJzTSEHu5Ck%2Fdk2pUHeH%2Fb9J1NbPr3xIOsvV9U%3D&reserved=0> ?

―

**Comment by Jeff-Tian** [comment]
PR https://github.com/gatsbyjs/gatsby/pull/37165 ready.
